sim/core/statweight.go

```diff
@@ -116,115 +114,40 @@ func CalcStatWeight(swr *proto.StatWeightsRequest, referenceStat stats.Stat, pro
-		SimOptions: simOptions,
-	}
-	baselineResult := RunRaidSim(baseSimRequest)
-	if baselineResult.ErrorResult != "" {
-		// TODO: get stack trace out.
-		return &StatWeightsResult{}
+	swr.SimOptions.UseLabeledRands = true
+
+	swBaseResponse := &proto.StatWeightRequestsData{
+		BaseRequest: &proto.RaidSimRequest{
+			Raid:       raidProto,
+			Encounter:  swr.Encounter,
+			SimOptions: swr.SimOptions,
+		},
+		EpReferenceStat: swr.EpReferenceStat,
+		StatSimRequests: []*proto.StatWeightsStatRequestData{},
 	}
 
-	var waitGroup sync.WaitGroup
-
 	// Do half the iterations with a positive, and half with a negative value for better accuracy.
-	resultsLow := make([]*proto.RaidSimResult, stats.UnitStatsLen)
-	resultsHigh := make([]*proto.RaidSimResult, stats.UnitStatsLen)
-
-	var iterationsTotal int32
-	var iterationsDone int32
-	var simsTotal int32
-	var simsCompleted int32
-
-	concurrency := (runtime.NumCPU() - 1) * 2
-	if concurrency <= 0 {
-		concurrency = 2
-	}
-
-	tickets := make(chan struct{}, concurrency)
-	for i := 0; i < concurrency; i++ {
-		tickets <- struct{}{}
-	}
-
-	doStat := func(stat stats.UnitStat, value float64, isLow bool) {
-		defer waitGroup.Done()
-		// wait until we have CPU time available.
-		<-tickets
-
-		simRequest := googleProto.Clone(baseSimRequest).(*proto.RaidSimRequest)
-		stat.AddToStatsProto(simRequest.Raid.Parties[0].Players[0].BonusStats, value)
-
-		reporter := make(chan *proto.ProgressMetrics, 10)
-		go RunSim(simRequest, reporter, nil) // RunRaidSim(simRequest)
-
-		var localIterations int32
-		var errorStr string
-		var simResult *proto.RaidSimResult
-
-		for metrics := range reporter {
-			atomic.AddInt32(&iterationsDone, metrics.CompletedIterations-localIterations)
-			localIterations = metrics.CompletedIterations
-			if metrics.FinalRaidResult != nil {
-				atomic.AddInt32(&simsCompleted, 1)
-				simResult = metrics.FinalRaidResult
-			}
-			if progress != nil {
-				progress <- &proto.ProgressMetrics{
-					TotalIterations:     atomic.LoadInt32(&iterationsTotal),
-					CompletedIterations: atomic.LoadInt32(&iterationsDone),
-					CompletedSims:       atomic.LoadInt32(&simsCompleted),
-					TotalSims:           atomic.LoadInt32(&simsTotal),
-				}
-			}
-			if metrics.FinalRaidResult != nil {
-				errorStr = metrics.FinalRaidResult.ErrorResult
-				break
-			}
-		}
-		// TODO: get stack trace out if final result error is set.
-		if errorStr != "" {
-			panic("Stat weights error: " + errorStr)
-		}
-
-		if isLow {
-			resultsLow[stat] = simResult
-		} else {
-			resultsHigh[stat] = simResult
-		}
-		tickets <- struct{}{}
-	}
-
 	const defaultStatMod = 40.0 // match to the impact of a single gem
 	statModsLow := make([]float64, stats.UnitStatsLen)
 	statModsHigh := make([]float64, stats.UnitStatsLen)
 
 	// Make sure reference stat is included.
-	statModsLow[referenceStat] = defaultStatMod
-	statModsHigh[referenceStat] = defaultStatMod
+	statModsLow[swr.EpReferenceStat] = -defaultStatMod
```

Comment:
Is this intended? The old code didn't have a negative value here

Reply:
I'm pretty sure that was a bug in the old code. The purpose of the low and high is to sim with a positive and negative mod. It probably never mattered because the reference stat is always included and added with a negative mod for the low case like all the other stats to sim.

Reply:
Yea I figured, but was just double checking 👍